tools/windows-node-installer/test/e2e/aws_test.go

```diff
@@ -1,90 +1,329 @@
+	t.Run("test instance is attache a Windows security group", checkInstanceIsAssociatedWithWindowsWorkerSg)
+	t.Run("test instance is associated with cluster worker's IAM", checkInstanceIsAssociatedWithClusterWorkerIAM)
+}
+
+// testDestroyWindowsInstance tests the deletion of a Windows instance and checks if the created instance and Windows
+// security group are deleted.
+func testDestroyWindowsInstance(t *testing.T) {
+	t.Run("test instance is terminated", destroyingWindowsInstance)
+	t.Run("test Windows security group is deleted", checkSgIsDeleted)
+	t.Run("test installer json file is deleted", checkInstallerJsonFileIsDeleted)
+}
+
+// setupAWSCloudProvider creates provider ofr Cloud interface and asserts type into AWS provider.
+// This is the first step of the e2e test and fails the test upon error.
+func setupAWSCloudProvider(t *testing.T) {
+	// The e2e test uses Microsoft Windows Server 2019 Base with Containers image, m4.large type, and libra ssh key.
+	cloud, err := cloudprovider.CloudProviderFactory(kubeconfig, awscredentials, "default", dir,
+		imageID, instanceType, sshKey)
 	if err != nil {
-		// TODO: Remove the lingering resources as part of test setup.
-		t.Fatalf("Error while creating a VM delete the virtual machine")
+		assert.FailNow(t, "unable to create an instance")
 	}
-	// Type Assert to AWS so that we can test other functionality
-	aws, ok := cloud.(*awscp.AwsProvider)
+
+	// Type assert to AWS so that we can test other functionality
+	provider, ok := cloud.(*awscp.AwsProvider)
 	if !ok {
-		t.Fatal("Error asserting cloudprovider to aws")
+		assert.FailNow(t, "Error asserting cloudprovider to awsProvider")
 	}
+	awsProvider = provider
+}
+
+// setupWindowsInstance creates a Windows instance and updates global information for infraID, createdInstanceID, and
+// createdSgID. All information updates are required to be successful or instance will be teared down.
+func setupWindowsInstance(t *testing.T) {
+	err := awsProvider.CreateWindowsVM()
+	if err != nil {
+		tearDownInstance(t, "error creating Windows instance", err)
+	}
+
+	infraID, err = awsProvider.GetInfraID()
+	if err != nil {
+		tearDownInstance(t, "error while getting infrastructure ID for the OpenShift cluster", err)
+	}
+
 	// Check instance and security group information in windows-node-installer.json.
 	info, err := resource.ReadInstallerInfo(dir + "/" + "windows-node-installer.json")
-	assert.NoError(t, err, "error reading windows-node-installer.json file")
-	assert.NotEmpty(t, info.SecurityGroupIDs, "security group is not created")
-	assert.NotEmpty(t, info.InstanceIDs, "instance is not created")
-	instance, err := aws.GetInstance(info.InstanceIDs[0])
 	if err != nil {
-		t.Fatalf("error getting ec2 instance object from instance ID %v", info.InstanceIDs[0])
+		tearDownInstance(t, "error reading from windows-node-installer.json file", err)
 	}
-	infraID, err := aws.GetInfraID()
+
+	if len(info.SecurityGroupIDs) > 0 && info.SecurityGroupIDs[0] != "" {
```

Comment:
What's the difference between this condition and https://github.com/openshift/windows-machine-config-operator/pull/49/files#diff-d69f962c88c5c87fe0fa39e99bde78b6R115